src/darwin/Framework/CHIP/MTRDevice_Concrete.mm

```diff
@@ -3319,6 +3319,133 @@ - (BOOL)_attributeDataValue:(NSDictionary *)one isEqualToDataValue:(NSDictionary
     return [one[MTRTypeKey] isEqual:theOther[MTRTypeKey]] && ((one[MTRValueKey] == theOther[MTRValueKey]) || [one[MTRValueKey] isEqual:theOther[MTRValueKey]]);
 }
 
+// _attributeDataValue:isEquivalentToDataValue: differs from
```

Comment:
Maybe this should be a free function instead of an MTRDevice method?

Reply:
I considered that... I basically did it by analogy with the _attributeDataValue:isEqualToDataValue: thing, but doing it as a free function is certainly doable.

Reply:
Making this a free function makes it hard to log the device, so not going to do that.